cpp/oneapi/dal/io/csv/common.cpp

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/io/csv/common.hpp"
+#include "oneapi/dal/exceptions.hpp"
+
+namespace oneapi::dal::csv {
+
+class detail::data_source_impl : public base {
+public:
+    char delimiter    = ',';
+    bool parse_header = false;
+    char *file_name   = nullptr;
+};
+
+using detail::data_source_impl;
+
+data_source::data_source(const char *file_name) : impl_(new data_source_impl{}) {
```

Comment:
Why not `std::string`, but `const char*`?

Reply:
If you care about dependencies on `std`, then we already have dependency on string in exceptions (in constructors of `std::invalid_argument`, `std::runtime_error`, etc.)